docs/design/proposals/resource-reference/resource-reference.md

```diff
@@ -80,27 +80,19 @@ of a field but also check if the Status is available etc...
 ```go
 type FieldConfig struct {
     ...
-    Reference *ReferenceConfig `json:"references,omitempty"`
+    Reference *ReferenceConfig `json:"reference,omitempty"`
     ...
 }
 
 // ReferenceConfig contains the instructions for how to add the referenced resource
 // configuration for a field.
 type ReferenceConfig struct {
-	// Kind mentions the K8s Kind of referenced resource
-	Kind string `json:"kind"`
-	// FieldPath refers to the the path of field which should be copied
-	// from referenced resource into TargetFieldName
-	FieldPath string `json:"field_path"`
-	// TargetFieldName is spec field name which gets value from
-	// referenced resource's FieldPath
-	TargetFieldName string `json:"target_field"`
-	// IsList mentions whether the new field should be AWSResourceReference or
-	// List of AWSResourceReference
-	IsList          bool   `json:"is_list"`
-	// Required mentions whether either of AWSResourceReference or
-	// TargetFieldName must be present in custom resource
-	Required        bool   `json:"required"`
+	// TargetResource mentions the K8s resource which is read to resolve the
+	// reference
+	TargetResource string `json:"target_resource"`
+	// SourcePath refers to the the path of field which should be copied
+	// to resolve the reference
+	SourcePath string `json:"source_path"`
```

Comment:
This should be SourceResource and SourcePath, no?

Reply:
Yes, you are right. Correcting now.

Reply:
Do you think we can drop the 'Source' from name?
The generator yaml will look like

```yaml
resource:
  Integration:
    fields:
      ApiId:
        references:
          resource: API
          path: Status.APIID
```

Reply:
> Do you think we can drop the 'Source' from name? @jaypipes The generator yaml will look like
> 
> ```yaml
> resources:
>   Integration:
>     fields:
>       ApiId:
>         references:
>           resource: API
>           path: Status.APIID
> ```

@vijtrip2 yes, absolutely